packages/block-library/src/image/edit.native.js

```diff
@@ -209,6 +220,28 @@ class ImageEdit extends React.Component {
 		];
 	}
 
+	onFocusCaption() {
+		if ( this.props.onFocus ) {
+			this.props.onFocus();
+		}
```

Comment:
This `onFocus()` call avoids two issues: 

1. Going from another `RichText` component to an image caption would result in the toolbar having the buttons for both blocks (testing scenario 8); and 
2. None of the styling toolbar buttons would appear when switching from an unsupported block to a non-empty image caption (testing scenario 9).  

Adding this same logic to the `onBlurCaption` to call `this.props.onBlur` did not appear to be strictly necessary because `this.props.onBlur` was always undefined in my testing, but I thought it was better to add it so that `onBlurCaption` and `onFocusCaption` would have consistent behavior.